content/blog/2017/12/2017-12-11-auto-convert-freestyle-jenkins-jobs-to-coded-devops-pipeline.adoc

```diff
@@ -0,0 +1,54 @@
+---
+layout: post
+title: "Auto-Convert Freestyle Jenkins Jobs to Coded DevOps Pipeline" 
+tags:
+- plugin
+- pipeline
+author: alyssat
+---
+
+[NOTE]
+====
+This is a guest post by Sanil Pillai, Director of Labs & Strategic Insights, Infostretch
+====
+
+link:https://www.infostretch.com[Infostretch] has created a link:https://plugins.jenkins.io/convert-to-pipeline[plugin] for DevOps teams migrating from Jenkins 1.0 to Jenkins 2.0 that automatically converts freestyle jobs to coded pipeline. This new plugin streamlines the process and accelerates pipeline on-boarding for any new set of applications. Previously, when migrating to Jenkins 2.0, converting freestyle jobs required DevOps engineers to drill down on each one of those hundreds or thousands of jobs to understand tools, configurations, URLs, parameters and more before rewriting them in pipeline syntax. This process is very manual, error-prone, lengthy and not cost-effective. Beyond saving time, the new plugin also assures adherence to proper coding standards and separates complex business logic and standards declaration from execution flow. 
+
+Key features:
+
+•	Convert single freestyle job to pipeline
+•	Convert chain of freestyle jobs to single pipeline
+•	Works with both open-source Jenkins and CloudBees Enterprise Jenkins
```

Comment:
Trademark violation 🐛

Reply:
@oleg-nenashev It's not the most helpful to make a comment like this without explaining why you believe this to be a trademark violation of the Jenkins trademark in a blog post on the Jenkins project's blog.

Reply:
"CloudBees Enterprise Jenkins" is not an approved trademark usage. "CloudBees Jenkins Enterprise" would be fine. https://wiki.jenkins.io/display/JENKINS/Approved+Trademark+Usage